### PR TITLE
nixVersions.nix_2_16: 2.16.2 -> 2.16.3

### PR DESCRIPTION
**Pull request by PerchunPak**
## Description of changes

nixd is undergoing a significant rewrite currently, so the latest release is probably stuck for a while on nix 2.16; however, that nix version has been pulled due to this CVE so currently you can't build it. 2.16.3 includes a patch for that CVE.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by wamserma — APPROVED** [review]

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -226,11 +226,8 @@ in lib.makeExtensible (self: ({
   };
 
   nix_2_16 = common {
-    version = "2.16.2";
-    hash = "sha256-VXIYCDkvAWeMoU0W2ZI0TeOszCZA1o8trz6YCPFD5ac=";
-    patches = [
-      patch-rapidcheck-shared
-    ];
```
Comment (PerchunPak) [inline pkgs/tools/package-management/nix/default.nix:233]: This patch was already applied in 2.16.3

https://github.com/NixOS/nix/commit/45ac34e90caf98e33a02c0eeb0e6b19d02ef0677

**Comment by wamserma** [comment]
`nixd` builds fine with https://github.com/NixOS/nixpkgs/commit/d1885787f6ab4c21b8ac0afcf994649ea7d78113 on x86_64-linux

**Comment by PerchunPak** [comment]
> `nixd` builds fine with [d188578](https://github.com/NixOS/nixpkgs/commit/d1885787f6ab4c21b8ac0afcf994649ea7d78113) on x86_64-linux

Can you share an overlay you used?

**Comment by wamserma** [comment]
> > `nixd` builds fine with [d188578](https://github.com/NixOS/nixpkgs/commit/d1885787f6ab4c21b8ac0afcf994649ea7d78113) on x86_64-linux
> 
> Can you share an overlay you used?

I had prepared the same PR against nixpkgs, the did `nix-build -K -A nixd` on my working copy. No overlay involved.

**Comment by wamserma** [comment]
Result of `nixpkgs-review pr 296670` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>nixVersions.nix_2_16</li>
    <li>nixVersions.nix_2_16.debug</li>
    <li>nixVersions.nix_2_16.dev</li>
    <li>nixVersions.nix_2_16.doc</li>
    <li>nixVersions.nix_2_16.man</li>
    <li>nixd</li>
  </ul>
</details>

**Pull request merged**